aspnetcore/fundamentals/servers/kestrel/endpoints.md

```diff
@@ -5,60 +5,111 @@ description: Learn about configuring endpoints with Kestrel, the cross-platform
+  Reverse proxy server examples include IIS, Nginx, and Apache. Hosting in a reverse proxy configuration requires [host filtering](xref:fundamentals/servers/kestrel/host-filtering).
 
-:::code language="csharp" source="~/fundamentals/servers/kestrel/samples/6.x/KestrelSample/Snippets/Program.cs" id="snippet_ConfigureEndpointDefaults":::
+* Host name `localhost` with port number or loopback IP with port number
 
-> [!NOTE]
-> Endpoints created by calling <xref:Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerOptions.Listen%2A> **before** calling <xref:Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerOptions.ConfigureEndpointDefaults%2A> won't have the defaults applied.
+  ```
+  http://localhost:5000/
+  http://127.0.0.1:5000/
+  http://[::1]:5000/
+  ```
+
+  When `localhost` is specified, Kestrel attempts to bind to both IPv4 and IPv6 loopback interfaces. If the requested port is in use by another service on either loopback interface, Kestrel fails to start. If either loopback interface is unavailable for any other reason (most commonly because IPv6 isn't supported), Kestrel logs a warning.
+
+Multiple URL prefixes can be specified by using a semicolon (`;`) delimiter:
+
+```
+http://*:5000;http://localhost:5001;https://hostname:5002
+```
 
-## Configure(IConfiguration)
+For more information, see [Override configuration](xref:fundamentals/host/web-host#override-configuration).
 
-Enables Kestrel to load endpoints from an <xref:Microsoft.Extensions.Configuration.IConfiguration>. The configuration must be scoped to the configuration section for Kestrel. The `Configure(IConfiguration, bool)` overload can be used to enable reloading endpoints when the configuration source changes.
+### Port 0
+
+When the port number `0` is specified, Kestrel dynamically binds to an available port. The following example shows how to determine which port Kestrel bound at runtime:
+
+:::code language="csharp" source="~/fundamentals/servers/kestrel/samples/6.x/KestrelSample/Snippets/Program.cs" id="snippet_IServerAddressesFeature":::
+
+Dynamically binding a port isn't available in some situations:
+
+* <xref:Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerOptions.ListenLocalhost%2A?displayProperty=nameWithType>
+* Binding TCP-based HTTP/1.1 or HTTP/2, and QUIC-based HTTP/3 together.
+
+### Specify ports only
+
+[!INCLUDE [http-ports](~/includes/http-ports.md)]
 
-By default, Kestrel configuration is loaded from the `Kestrel` section and reloading changes is enabled:
+### HTTPS URL prefixes
+
+HTTPS URL prefixes can be used to define endpoints only if a default certificate is provided in the HTTPS endpoint configuration. For example, use <xref:Microsoft.AspNetCore.Server.Kestrel.KestrelServerOptions> configuration or a configuration file, as shown [later in this article](#configure-certificates-in-appsettingsjson).
```

Comment:
Is that true?  I think you can pick up a default cert from the cert store?

Reply:
What should we say here, then?